### ISSUE TITLE
Popularity is broken when the wikidata entry points to a wikipedia redirect

### ISSUE THREAD

**Issue by davidebbo** [OneZoom/tree-build#59]
[This is a spin-off issue from #49]

Another weird case: [Leopardus pajeros](https://www.wikidata.org/wiki/Q311417) is the Pampas cat.

Problem is that the wikidata entry has the english link linking to Leopardus_pajeros (which is a redirect), instead of to the main Pampas cat page: `"enwiki": { "title": "Leopardus pajeros" }`. So we end up looking up `Leopardus pajeros` in the Page Count file, and not finding anything, because all the hits are with the Pampas cat entry.

This is a completely different case from above, as it doesn't relate to needing to follow wikidata synonyms.

**Comment by davidebbo** [comment]
From @hyanwong:

>  has the english link linking to Leopardus_pajeros (which is a redirect)

So we can either change the link in Wikidata and/or somehow keep redirects in the wikipedia dump, and follow those?

I seem to remember that I did do a bit of redirect-following in my original wikimedia parser, but I think I didn't think about that case on wikipedia.

Yan

**Comment by davidebbo** [comment]
> change the link in Wikidata

Interestingly, if you look at the links on https://www.wikidata.org/wiki/Q311417, next to 'en', there is a little arrow symbol. If you hover over it, it says "**intentional** sitelink to redirect". I'm not sure what the reasoning is, as in most cases, the link is to the main page with no redirects.

> somehow keep redirects in the wikipedia dump, and follow those

The wikiDATA dump doesn't know about redirects afaik. It just has wikiPEDIA links, which sometimes happen to redirect. So I'm not sure if we can get the redirect information, other than by actually making http requests to it, which is too painful/slow.

**Comment by davidebbo** [comment]
From @hyanwong:

> The wikiDATA dump doesn't know about redirects afaik. It just has wikiPEDIA links, which sometimes happen to redirect. So I'm not sure if we can get the redirect information, other than by actually making http requests to it, which is too painful/slow.

Indeed, but perhaps the `enwiki-latest-page.sql.gz` file contains information on redirects?

**Comment by davidebbo** [comment]
> Indeed, but perhaps the `enwiki-latest-page.sql.gz` file contains information on redirects?

Yes, it has a `page_is_redirect` Boolean field (https://www.mediawiki.org/wiki/Manual:Page_table#page_is_redirect). I don't think it gives the redirection target, but at least if we had to make a request, that would greatly reduce the number of cases where it's needed.

**Comment by hyanwong** [comment]
> that would greatly reduce the number of cases where it's needed.

Good point. This makes the logic a bit more convoluted, doesn't it, but I think it is probably worth doing. As a half-way house we could check in the SQL dump but simply emit a warning if it appears as if the name is a redirect.

**Comment by hyanwong** [comment]
> Today, everything happens offline, which is nice.

Yes, I agree this is much nicer.

**Comment by hyanwong** [comment]
Great, thanks @davidebbo. 10334 sounds quite a lot, but as you say, only a subset will be relevant to us, fortunately.
